packages/openactive-integration-tests/test/helpers/request-state.js

```diff
@@ -207,14 +207,33 @@ class RequestState {
     return isResponse(this.c1Response);
   }
 
+  /**
+   * As leasing is optional, getting the totalPaymentDue should not be dependent on C1 or C2 being completed.
```

Comment:
Although Leasing is optional C1 and C2 are not (ref: https://openactive.io/open-booking-api/EditorsDraft/#broker-contract-with-booking-system)

So `totalPaymentDue` from the response must always be used. See also https://github.com/openactive/open-booking-api/issues/152

Reply:
Ah ok, in that case the RefImpl needs to be changed as it returns a PaymentMismatchError when it shouldn't